tagstudio/src/qt/widgets/media_player.py

```diff
@@ -0,0 +1,207 @@
+
+        self.setFixedHeight(50)
+
+        self.filepath: Path | None = None
+        self.player = QMediaPlayer()
+        self.player.setAudioOutput(QAudioOutput(QMediaDevices().defaultAudioOutput(), self.player))
+
+        # Used to keep track of play state.
+        # It would be nice if we could use QMediaPlayer.PlaybackState,
+        # but this will always show StoppedState when changing
+        # tracks. Therefore, we wouldn't know if the previous
+        # state was paused or playing
+        self.is_paused = False
+
+        # Subscribe to player events from MediaPlayer
+        self.player.positionChanged.connect(self.player_position_changed)
+        self.player.mediaStatusChanged.connect(self.media_status_changed)
+        self.player.playingChanged.connect(self.playing_changed)
+        self.player.audioOutput().mutedChanged.connect(self.muted_changed)
+
+        # Media controls
+        self.base_layout = QGridLayout(self)
+        self.base_layout.setContentsMargins(0, 0, 0, 0)
+        self.base_layout.setSpacing(0)
+
+        self.pslider = QSlider(self)
+        self.pslider.setFocusPolicy(Qt.FocusPolicy.StrongFocus)
+        self.pslider.setTickPosition(QSlider.TickPosition.NoTicks)
+        self.pslider.setSingleStep(1)
+        self.pslider.setOrientation(Qt.Orientation.Horizontal)
+
+        self.pslider.sliderReleased.connect(self.slider_released)
+        self.pslider.valueChanged.connect(self.slider_value_changed)
+
+        self.media_btns_layout = QHBoxLayout()
+
+        policy = QSizePolicy(QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed)
+
+        self.play_pause = QPushButton("", self)
+        self.play_pause.setFlat(True)
+        self.play_pause.setSizePolicy(policy)
+        self.play_pause.clicked.connect(self.toggle_pause)
+
+        self.load_play_pause_icon(playing=False)
+
+        self.media_btns_layout.addWidget(self.play_pause)
+
+        self.mute = QPushButton("", self)
+        self.mute.setFlat(True)
+        self.mute.setSizePolicy(policy)
+        self.mute.clicked.connect(self.toggle_mute)
+
+        self.load_mute_unmute_icon(muted=False)
+
+        self.media_btns_layout.addWidget(self.mute)
+
+        self.position_label = QLabel("positionLabel")
+        self.position_label.setAlignment(Qt.AlignmentFlag.AlignRight)
+
+        self.base_layout.addWidget(self.pslider, 0, 0, 1, 2)
+        self.base_layout.addLayout(self.media_btns_layout, 1, 0)
+        self.base_layout.addWidget(self.position_label, 1, 1)
+
+    def format_time(self, ms: int) -> str:
+        """Format the given time.
+
+        Formats the given time in ms to a nicer format.
+
+        Args:
+            ms: Time in ms
+
+        Returns:
+            A formatted time:
+
+            "1:43"
+
+            The formatted time will only include the hour if
+            the provided time is at least 60 minutes.
+        """
+        time = gmtime(ms / 1000)
+        fmt = "%-H:%-M:%S" if time.tm_hour > 0 else "%-M:%S"
```

Comment:
The `-` in `%-H:%-M:%S` and `%-M:%S` isn't supported on all platforms. It should be removed or you can use extra logic to manually strip the leading zeros.

Reply:
looks good.